### ISSUE TITLE
Visualization stops runtime and does not generate output

### ISSUE THREAD

**Issue by jkoura** [MaartenGr/BERTopic#1815]
Hi, firstly thank you for all your work with BERTopic!

I am using it for my thesis and I am stuck on visualizing the documents. 
I am using colab pro and when observing my resources, there is not a spike that indicates massive memory usage before it stops. Sometimes, it just executes, but no visualization comes out and no error message.  I tried your examples using the abstracts example on a different notebook and it worked just fine. Now my project has a lot more documents. I've taken the following steps:

I reduced the embeddings first:

`reduced_embeddings = UMAP(n_neighbors=10, n_components=2, min_dist=0.0, metric='cosine', random_state=42).fit_transform(embeddings)`

I ran my topic model:

`topic_model = BERTopic(
    # Submodels
    embedding_model=embedding_model,
    representation_model=representation_model,
    umap_model=umap_model,
    top_n_words=10,
    nr_topics=50,
    verbose=True).fit(docs)`

finally this:

`topic_model.visualize_documents(docs, reduced_embeddings=reduced_embeddings) `

and this

```
from matplotlib import pyplot as plt
from umap import UMAP

topic_model.visualize_documents(
    docs_list,
    reduced_embeddings=reduced_embeddings,
    topics=[2,3],
    height=600
)
```

Both lead to my runtime stopping and reconnecting at the same time, but no output is created and no error message.
I would be grateful for any help!

**Comment by MaartenGr** [comment]
Most likely, you have simply to many datapoints to be properly rendered with your current RAM capacity. Either switch to high RAM or using the `sample` parameter in `.visualize_documents` to only plot a sample of your documents. If you have millions of documents, there is often no need to visualize them all since there is likely to be some redundancy amongst the embeddings.

**Comment by jkoura** [comment]
It worked, thank you for your reply!
